tests/integration/test_integration.py

```diff
@@ -1755,3 +1755,27 @@ def test_groupby_aggregation_within_with_failures(
 
 def test_diff_average_between():
     return
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        (identity, "col_int", "col_int", None, 1.0),
+        (identity, "col_int", "col_int", Condition("col_int >= 3"), 1.0),
+    ],
+)
+def test_ks_2sample_constraint_perfect_between(engine, int_table1, int_table2, data):
+    """
+    Test Kolmogorov-Smirnov for the same column -> p-value should be perfect 1.0.
+    """
+    (operation, col_1, col_2, condition, significance_level) = data
+    req = requirements.BetweenRequirement.from_tables(*int_table1, *int_table1)
```

Comment:
Should the second argument be `int_table2`? If not, we could remove it from the function parameters.

Reply:
I wanted to test the same data twice to get a perfect p_value of 1.0, and therefore I need the same tables.
But you're right; the second argument is not needed!